### PR TITLE
Add options for -n or --numeric-gid-uid.

### LINKED ISSUE #807: Show numeric UID and GID
Please add a parameter equivalent to ls' `-n, --numeric-uid-gid` parameter that lists the user and group with their numeric IDs instead of the names.

This is especially pretty handy when it comes to debugging file access with docker containers, as user-names in the container might be mapped to different numeric IDs on the host.

### PR DESCRIPTION
**Pull request by b05902132**
Closes #807.

This option is only available in -l.
There's test for rendering, but no test for option parsing.
(I don't understand that section of code.)

Also in `ls` this option implies `-l`, is it worth it to follow behavior of `ls`?

### CODE REVIEW

File: completions/completions.fish [hunk 1]
```diff
@@ -65,6 +65,7 @@ complete -c exa -s 't' -l 'time'  -x -d "Which timestamp field to list" -a "
     created\t'Display created time'
 "
 complete -c exa -s 'm' -l 'modified'      -d "Use the modified timestamp field"
+complete -c -exa -s 'n' -l 'numeric-uid-gid' -d "List numeric user and group IDs."
```
Comment (ariasuni) [inline completions/completions.fish:68]:
```suggestion
complete -c exa -s 'n' -l 'numeric-uid-gid' -d "List numeric user and group IDs."
```

File: src/options/help.rs [hunk 2]
```diff
@@ -46,6 +46,7 @@ LONG VIEW OPTIONS
   -H, --links          list each file's number of hard links
   -i, --inode          list each file's inode number
   -m, --modified       use the modified timestamp field
+  -n, --numeric-uid-gid list numeric user and group IDs
```
Comment (ariasuni) [inline src/options/help.rs:49]: There should two spaces between the option name and description:

```suggestion
  -n, --numeric-uid-gid  list numeric user and group IDs
```

This will be displayed with `exa --help`. Please align the descriptions to the same column.

File: src/options/view.rs [hunk 3]
```diff
@@ -265,6 +266,14 @@ impl TimeFormat {
     }
 }
 
+impl UserFormat {
+    fn deduce(matches: &MatchedFlags<'_>) -> Result<Self, OptionsError> {
+        let flag = matches.has(&flags::NUM_UGID)?;
+        Ok(if flag { Self::Numeric } else { Self::Name })
+    }
+
+}
```
Comment (ariasuni) [inline src/options/view.rs:275]: Just a style nitpicking:

```suggestion
}


impl UserFormat {
    fn deduce(matches: &MatchedFlags<'_>) -> Result<Self, OptionsError> {
        let flag = matches.has(&flags::NUM_UGID)?;
        Ok(if flag { Self::Numeric } else { Self::Name })
    }
}
```

**Review by ariasuni — CHANGES_REQUESTED** [review]
The current behavior is correct and `-n` shouldn’t imply `-l`: in exa, options relative to a specific mode (long, grid, etc.) are ignored in different modes to allow overriding aliases.

I didn’t test your change, but the code looks good!

**Comment by b05902132** [comment]
Thanks for your review! I applied all your suggested chanes. It's the first time I use this feature; not sure if I did it correctly though.

File: src/options/help.rs (hunk 2 above)
Comment (b05902132) [reply src/options/help.rs]: `--numeric-uid-gid` is too long to make the description align with other descriptions. Adding more spaces makes the alignment off even more. I will add a space to other options to make their descriptions align the description of `-n`.

**Comment by b05902132** [comment]
Hi. I ran `exa --help` and found that the suggested change actually made the alignment worse. I added a new commit to fix that. 
Also I'd like to squash all suggested changes into one commit, but I'm new to open source world and not sure if it's allowed. I heard that you should not change published commits, but I don't think any one will base their work on my branch.

**Review by ariasuni — CHANGES_REQUESTED** [review]
You should also add the option to the list of options that are useless without long for strict mode:
https://github.com/ogham/exa/blob/435161e6e4436f10e4faaa03c5f3deef26e12b36/src/options/view.rs#L87-L92

---

For tests, you should add the options in:
https://github.com/ogham/exa/blob/435161e6e4436f10e4faaa03c5f3deef26e12b36/src/options/view.rs#L343-L348

You can then copy
https://github.com/ogham/exa/blob/435161e6e4436f10e4faaa03c5f3deef26e12b36/src/options/view.rs#L553

and
https://github.com/ogham/exa/blob/435161e6e4436f10e4faaa03c5f3deef26e12b36/src/options/view.rs#L564
but use `-n` or `--numeric-uid-gid` instead.

**Comment by ogham** [comment]
Hi, thanks for this! I can see the use for this option.

My feeling is that I'd rather call it just `--numeric` rather than `--numeric-uid-gid`. I can't think of any other columns where we display a name based on a number other than the user and group, and users will expect options with long names and hyphens in to be more complex, when this one is quite simple. There's also precedent: netstat uses `-n`, and GNU netstat uses `--numeric`.

Don't worry about the help text; keeping things in alignment is tricky when multiple people are making edits to the same file. It happens in code too. I'll sort it out when everything settles.

**Comment by ogham** [comment]
> (I don't understand that section of code.)

Yeah, it was written when I was feeling more, uh... artistic. When I looked back on it recently I wished I'd just kept it simple. So don't worry if it was hard to follow.

> Also I'd like to squash all suggested changes into one commit, but I'm new to open source world and not sure if it's allowed. I heard that you should not change published commits, but I don't think any one will base their work on my branch.

There's no rule that applies to all projects, but _in general_, your second point is correct: if multiple people are using a branch then you should leave it alone, but if it's your own development branch that has not been merged yet, then feel free to squash.

**Comment by b05902132** [comment]
> Hi, thanks for this! I can see the use for this option.
> 
> My feeling is that I'd rather call it just `--numeric` rather than `--numeric-uid-gid`. I can't think of any other columns where we display a name based on a number other than the user and group, and users will expect options with long names and hyphens in to be more complex, when this one is quite simple. There's also precedent: netstat uses `-n`, and GNU netstat uses `--numeric`.
It was following GNU `ls`. Since `exa` does not need to follow `ls`, I've updated the option to be `--numeric`.
> 
> Don't worry about the help text; keeping things in alignment is tricky when multiple people are making edits to the same file. It happens in code too. I'll sort it out when everything settles.
Okay.
